packages/components/src/input/SaltInput.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from "react";
+import { TextInput } from "../input/";
+import { wordsToSalt } from "@joincivil/utils";
+
+export interface SaltInputProps {
+  name: string;
+  label?: string;
+  placeholder?: string;
+  onChange(name: string, value: string): void;
+}
+
+export interface SaltInputState {
+  isValid: boolean;
+  value: string;
+}
+
+export class SaltInput extends React.Component<SaltInputProps, SaltInputState> {
+  constructor(props: SaltInputProps) {
+    super(props);
+    this.state = {
+      isValid: false,
+      value: "",
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  public render(): JSX.Element {
+    const { name, label, placeholder } = this.props;
+    const { value, isValid } = this.state;
+
+    return <TextInput label={label} placeholder={placeholder} name={name} onChange={this.handleChange} value={value} />;
+  }
+
+  private handleChange(name: string, value: string): void {
```

Comment:
You could probably do this and get rid of the `this.handleChange.bind()` line in the constructor.

(Which would match our coding style thusfar).

```
private const handleChange = (name: string, value: string): void => {
 ...
}
```